project-handbook/license-review.md

```diff
@@ -0,0 +1,100 @@
+
+## 1. Find license(s) of and inside own code
+
+To find the license of the own code and potential secondary licenses, you can use and mix different steps:
+
+### a) Look at License files
+
+* Take a look files such as `LICENSE`, `COPYING` or inside a `LICENSES` folder.
+* Check out what's written inside of a `README` or `CONTRIBUTING` file.
+
+### b) Search for license statements: manually
+
+Search for licenses using these commands:
+
+* `git grep -i "licen[s|c]"`
+* `git grep -iE "copy(right|left)"`
+* `git grep -i "public domain"`
+
+Analyse the findings for licensing/copyright by third parties. Note down these licenses and include them into their consideration.
+
+### c) Search for license statements: automatically
+
+You may a tool like the [SBOM Workbench](https://github.com/scanoss/sbom-workbench) by ScanOSS. This may help you finding re-used snippets, e.g. from StackOverflow, which introduce other licenses and copyrights.
+
+### d) Ask the project team
+
+An honest discussion with the project team might also reveal the use of code from third parties, e.g. by external contractors or from company-internal projects. Make sure to understand the history of the project.
+
+## 2. Licensing of dependencies
+
+### a) Create SBOM
+
+One of these two steps are recommended and usually return the same results:
+
+1. Using compliance-assistant: `compliance-assistant sbom generate -d /path/to/code -o sbom.json`
+2. Using syft: `syft scan /path/to/code -o cyclonedx-json > sbom.json`
+
+### b) Enrich SBOM
+
+Enrich the SBOM with quite accurate licensing data from [ClearlyDefined](https://clearlydefined.io/), and creates a new file:
+
+`compliance-assistant sbom enrich -f sbom.json -o sbom-enriched.json`
+
+### c) Extract and evaluate licensing data
+
+Next, we extract and unify all contained Open Source licenses:
+
+`compliance-assistant licensing list -f sbom-enriched.json -o plain`
+
+Example output:
+
+```
+0BSD
+Apache-2.0
+Apache-2.0 AND MIT
+Apache-2.0 OR MIT
+BSD-2-Clause
+BSD-3-Clause
+BSD-3-Clause OR GPL-2.0-only
+BlueOak-1.0.0
+CC-BY-3.0
+CC-BY-4.0
+CC0-1.0
+ISC
+ISC AND MIT
+LGPL-3.0-or-later
+MIT
+Python-2.0
+Unlicense
+```
+
+### d) Optional: Conclude a valid outbound license
+
+This is not easy to automatically evaluate, and tools often have a hard time putting a larger number of licenses together. So this is optional.
+
+`compliance-assistant licensing outbound -f sbom-enriched.json`
+
+
+### e) Optional: Check for security issues
+
+We can use [grype](https://github.com/anchore/grype) to check the dependencies for known security vulnerabilities. May be useful to assess potential security risks:
+
+`grype --add-cpes-if-none sbom:sbom-enriched.json`
+
+## 3. Evaluate found licenses
+
+Whether all found licenses fit together or introduce problems has to be evaluated manually. Currently, we don't have a way to define that automatically.
```

Comment:
Ultimately, I think this license review has to be done by the TC, from the data gathering to the analysis. This way, we can ensure that the licensing situation is understandable and clear. Perhaps it makes sense to lower this threshold for the sandbox, but at least for stage 1 it makes sense IMHO.

Regarding the interval of such checks, I think the minimum would be to make such a review on each stage move, as the project probably gained more complexity. Ideally, projects would set up a suitable licensing policy and check that regularly.

Reply:
This would also be addressed by making this document here a general document and add the procedural information in the documentation of the incubation process.

Reply:
See resolution in https://github.com/OpenRailAssociation/technical-committee/pull/140/files#r1765241258